system_config/scripts/replace_params_yaml.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+# SPDX-License-Identifier: MIT
+
+# Exits immediately on failure.
+set -eu
+
+# Function to display usage information
+usage() {
+    echo "Usage: $0 <-f|--file FILE> <-p|--param PARAM_NAME="PARAM_VALUE"> [-p|--param PARAM_NAME="PARAM_VALUE"]..."
+    echo "Example:"
+    echo "  $0 -f /path/to/file/config.yaml -p param1=value1 -p param2=value2..."
+}
+
+# Create param list
+param_list=()
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]
+do
+    key="$1"
+
+    case $key in
+        -f|--file)
+            file="$2"
+            shift # past argument
+            shift # past value
+            ;;
+        -p|--param)
+            param_list+=("$2")
+            shift # past argument
+            shift # past value
+            ;;
+        -h|--help)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "Unknown argument: $key"
+            usage
+            exit 1
+    esac
+done
+
+# Check if all required arguments have been set
+if [[ -z "${file}" || ${#param_list[@]} -eq 0 ]]; then
+    echo "Error: Missing required arguments:"
+    [[ -z "${file}" ]] && echo "  -f|--file"
+    [[ ${#param_list[@]} -eq 0 ]] && echo "  -p|--param"
+    echo -e "\n"
+    usage
+    exit 1
+fi
+
+# Uncomment any commented out entries so the file can be properly processed
```

Comment:
in fact, is this step even necessary? it looks like the script below does a lot of work to clean up after itself if stuff doesn't get populated. would it be simpler to leave these as comments and just insert values that are passed to this script? we might not need the cleanup from line 72 if we do it that way

Reply:
looking at later files, it seems these extra steps might necessary because some placeholder values are in comments and others aren't. I'd suggest that we just put every placeholder value in a comment, then we don't have to uncomment them or remove items that weren't updated with this script (so lines 58 and 72 could be deleted, and line 67 would be updated to append rather than replace in the file). this could result in unused config items getting added to the file, but parsers should ignore that (we should of course confirm) and it's easier to extend configs with new values

Reply:
I will give your comments some thought. for the meantime I simplified the regex similar to what you suggested and updated the comment to add that non-escaped regex for simpler parsing.

Reply:
I see that the value of enforcing a consistent template format and that generating the configuration could lead to a marginally simpler script. Considering our services currently have different template formats I would want to leave this as is for now as it makes the path to blueprint set up quicker. I think this is something that can be revisited though in the future.